aten/src/ATen/NamedTensorUtils.cpp

```diff
@@ -391,10 +391,8 @@ void propagate_names_for_expand(const Tensor& result, const Tensor& self) {
     return;
   }
   std::vector<Dimname> outnames(result_dim, Dimname::wildcard());
-  std::copy(
-      self.opt_names()->begin(),
-      self.opt_names()->end(),
-      outnames.begin() + result_dim - self.dim());
+  auto names = self.names();
```

Comment:
nit: is this doing a copy? It can be a const ref, right?

Reply:
Can be